pac4j-core/src/main/java/org/pac4j/core/util/CommonHelper.java

```diff
@@ -226,20 +242,83 @@ public static String toString(final Class<?> clazz, final Object... args) {
      * @return the input stream
      */
     public static InputStream getInputStreamFromName(String name) {
-        if (name.startsWith(RESOURCE_PREFIX)) {
-            String path = name.substring(RESOURCE_PREFIX.length());
-            if (!path.startsWith("/")) {
-                path = "/" + path;
-            }
-            return CommonHelper.class.getResourceAsStream(path);
-        } else {
-            try {
-                return new FileInputStream(name);
-            } catch (FileNotFoundException e) {
-                throw new TechnicalException(e);
-            }
-        }
-    }
+		int prefixEnd = name.indexOf(":");
+		String prefix = null;
+		String path = name;
+		if (prefixEnd != -1) {
+			prefix = name.substring(0, prefixEnd);
+			path = name.substring(prefixEnd + 1);
+		}
+		if (prefix == null) {
+			throw new TechnicalException("prefix is not defined");
+		}
+
+		switch (prefix) {
+		case RESOURCE_PREFIX:
+			if (!path.startsWith("/")) {
+				path = "/" + path;
+			}
+			// The choice here was to keep legacy behavior and remove / prior to
+			// calling classloader.getResourceAsStream.. or make it work exactly
+			// as it did before but have different behavior for resource: and
+			// classpath:
+			// My decision was to keep legacy working the same.
+			return CommonHelper.class.getResourceAsStream(path);
+		case CLASSPATH_PREFIX:
+			return Thread.currentThread().getContextClassLoader().getResourceAsStream(path);
+		case HTTP_PREFIX:
+			logger.warn("IdP metadata is retrieved from an insecure http endpoint [{}]", path);
+			return getInputStreamViaHttp(name);
+		case HTTPS_PREFIX:
+			return getInputStreamViaHttp(name);
+		default:
+			try {
+				return new FileInputStream(path);
+			} catch (FileNotFoundException e) {
+				throw new TechnicalException(e);
+			}
+		}
+
+	}
+
+	private static InputStream getInputStreamViaHttp(String name) {
+		URLConnection con = null;
+		try {
```

Comment:
http://kingori.co/minutae/2013/04/httpurlconnection-disconnect/ and http://stackoverflow.com/questions/4767553/safe-use-of-httpurlconnection.. Basically this code is the same as it is in spring.. When the inputstream is close it can free up resources and depending on keep-alive setting will do the disconnect as far as I understand it.